src/reducers/vis-state-merger.js

```diff
@@ -111,13 +110,24 @@ export function mergeFilters(state, filtersToMerge) {
   // flatten all filter dataIds
   const datasetsToFilter = uniq(flattenDeep(merged.map(f => f.dataId)));
 
-  const filtered = applyFiltersToDatasets(datasetsToFilter, updatedDatasets, updatedFilters);
+  const filtered = applyFiltersToDatasets(datasetsToFilter, updatedDatasets, updatedFilters, state.layers);
```

Comment:
Can you add a test case to `vis-state-merger-test.js` with a map saved with a polygon filter, some other filter and a layer with `colorField`, then loaded it back to test this?